Atomic/backends/_containers_storage.py

```diff
@@ -84,20 +88,79 @@ def update(self, name, **kwargs):
         """
         raise UnderDevelopment()
 
+    # This needs to be fixed if the notion of dangling becomes real
+    # for containers-image.
+
+    def _make_container(self, container, con_struct, deep=False):
+        con_obj = Container(container, backend=self)
+        con_obj.id = con_struct['id']
+        con_obj.image = con_struct['image_id']
+        con_obj.image_name = con_struct['image']
+        con_obj.created = time.mktime(time.strptime(con_struct['createdAt'].split(".")[0], "%Y-%m-%dT%H:%M:%S"))
+        con_obj.status = con_struct['status']
+        con_obj.state = con_obj.status.split()[0]
+        con_obj.name = con_struct['names']
+        con_obj.labels = con_struct['labels']
+        con_obj.running = True if con_obj.status.lower().startswith("up") else False
+        con_obj.runtime = "runc"
+
+        return con_obj
+
     def get_containers(self):
         """
         Get containers for the backend
         :return: list of container objects
         """
-        raise UnderDevelopment()
+        containers = json.loads(util.kpod("ps -a --format json"))
+        container_objects = []
+        for container in containers:
+            container_objects.append(self._make_container(container['id'], container))
+        return container_objects
+
+    def _make_image(self, image, img_struct, deep=False, remote=False):
+        img_obj = Image(image, remote=remote)
+        img_obj.backend = self
+        if not remote:
+            img_obj.id = img_struct['id']
+            img_obj.repotags = img_struct['names']
+            img_obj.created = DT.datetime.strptime(img_struct['created'], "%b %d, %Y %H:%M").strftime("%Y-%m-%d %H:%M")
+            img_obj.size = img_struct['size']
+            img_obj.virtual_size = img_obj.size
+            img_obj.version = img_obj.version
+            img_obj.digest = img_struct['digest']
+            # img_obj.original_structure = img_struct
+
+        #if deep:
+            # img_obj.deep = True
+            # img_obj.repotags = img_struct['RepoTags']
+            # img_obj.config = img_struct['Config'] or {}
+            # img_obj.labels = img_obj.config.get("Labels", None)
+            # img_obj.os = img_struct['Os']
+            # img_obj.arch = img_struct['Architecture']
+            # img_obj.graph_driver = img_struct['GraphDriver']
+            # img_obj.version = img_obj.get_label('Version')
+            # img_obj.release = img_obj.get_label('Release')
+            # img_obj.parent = img_struct['Parent']
+            # img_obj.original_structure = img_struct
+            # img_obj.cmd = img_obj.original_structure['Config']['Cmd']
+        return img_obj
 
     def get_images(self, get_all=False):
         """
         Get images for the backend
         :param get_all: bool
         :return:  list of image objects
         """
-        raise UnderDevelopment()
+        images = json.loads(util.kpod("images --format json"))
```

Comment:
This feels like it should be a list rather then a string.

Reply:
util.kpod(["images", "--format", "json"])

Reply:
sure, if you prefer ... will fix up this and other instances.